common-build.gradle

```diff
@@ -47,3 +47,133 @@ task syncVersionAndroid  {
     }
 }
 
+task generateIOSBindings {
+    doLast() {
+        // Only generate bindings for iOS projects
+        if ( !project.hasProperty('iosSdkModuleName') ) {
+            return
+        }
+
+        println("Updating bindings for ${iosSdkModuleName} module.")
+
+        // clone iOS SDK at specified version, if it doesn't already exist
+        println("Using iOS SDK version ${airshipProperties.iosVersion}")
+        def tmpDir = System.getProperty("java.io.tmpdir")
+        def iosTempRepo = new File("${tmpDir}/ios-library-for-xamboni")
+        iosTempRepo.deleteOnExit()
+
+        boolean cloneRepo = false
+        if ( !iosTempRepo.exists() ) {
+            // cloned repo doesn't exist, so clone it
+            cloneRepo = true
+        } else {
+            // cloned repo exists, so check if it is the right version
+            def out = new ByteArrayOutputStream()
+            exec {
+                commandLine "git", "-C", "${iosTempRepo}", "tag", "--points-at", "HEAD"
```

Comment:
This is ok, but I think we could use https://blog.gradle.org/introducing-source-dependencies to have gradle pull dependencies for us

Reply:
I think this can be done, but it looks like a fair amount of work to learn how to do it and implement. I'm not sure it is worth it at this time.